applications/crossbar/src/modules/cb_auth.erl

```diff
@@ -372,3 +412,121 @@ add_app(Context) ->
 -spec add_provider(cb_context:context()) -> cb_context:context().
 add_provider(Context) ->
     Context.
+
+-spec reset_system_identity_secret(cb_context:context()) -> cb_context:context().
+reset_system_identity_secret(Context) ->
+    case kz_auth_identity:reset_system_secret() of
+        {'ok', _} -> cb_context:set_resp_status(Context, 'success');
+        {'error', _Reason} ->
+            lager:warning("failed to reset system identity secret: ~p", [_Reason]),
+            cb_context:add_system_error('datastore_fault', Context)
+    end.
+
+-spec reset_identity_secret(cb_context:context()) -> cb_context:context().
+reset_identity_secret(Context) ->
+    OwnerId = kz_json:get_ne_binary_value(<<"owner_id">>, cb_context:doc(Context)),
```

Comment:
use cb_context:owner_id , getting owner_id from the doc does not ensure the path